src/index.js

```diff
@@ -52,25 +54,52 @@ function main(client, options, _performance) {
 
 		const promises = [];
 
-		functions.forEach(function (key) {
-			const contextKey = toContextKey(key);
-			if (subset && subset.indexOf(contextKey) > -1) {
-				return;
-			}
-
-			if (key in evolv.javascript.variants) {
-				let promise = MiniPromise.createPromise(function(resolve, reject) {
-					try {
-						if (!evolv.javascript.variants[key].call({key: contextKey}, resolve, reject)) {
-							resolve();
-						}
-					} catch(err) {
-						reject(err);
-					}
-				});
-				promises.push(promise);
-			}
-		});
+    const immediateFunctions = functions.filter(isImmediate(evolv.javascript.variants));
```

Comment:
Do the function arrays remain correctly sorted so context will always execute before variant in the group?